pyartifactory/objects.py

```diff
@@ -231,6 +169,91 @@ def unlock(self, name: str) -> None:
         logger.debug("User % successfully unlocked", name)
 
 
+class ArtifactoryPermission(ArtifactoryObject):
+    """Models an artifactory permission."""
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super().__init__(artifactory)
+        if self._api_version == 2:
+            self._uri = "v2/security/permissions"
+        if self._api_version == 1:
+            self._uri = "security/permissions"
+
+    def create(
+        self, permission: Union[Permission, PermissionV2]
+    ) -> Union[Permission, PermissionV2]:
+        """
+        Creates a permission
+        :param permission: Permission v2 or v1 object
+        :return: Permission v2 or v1
+        """
+        permission_name = permission.name
+        try:
+            self.get(permission_name)
+            logger.debug("Permission %s already exists", permission_name)
+            raise PermissionAlreadyExistsException(
+                f"Permission {permission_name} already exists"
+            )
+        except PermissionNotFoundException:
+            self._put(f"api/{self._uri}/{permission_name}", json=permission.dict())
+            logger.debug("Permission %s successfully created", permission_name)
+            return self.get(permission_name)
+
+    def get(self, permission_name: str) -> Union[Permission, PermissionV2]:
+        """
+        Read permission from artifactory. Fill object if exist
+        :param permission_name: Name of the permission to retrieve
+        :return: Permission
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{permission_name}")
+            logger.debug("Permission %s found", permission_name)
+            return (
+                Permission(**response.json())
+                if self._artifactory.api_version == 1
+                else PermissionV2(**response.json())
+            )
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 or error.response.status_code == 400:
+                logger.error("Permission %s does not exist", permission_name)
```

Comment:
Not sure but i think 400 doesn't necessarily mean "does not exist".

Reply:
Normally it doesn't, but the Artifactory API sometimes sends a 400 error code. That's why we added it.